lib/SILGen/SILGenApply.cpp

```diff
@@ -4239,7 +4247,9 @@ namespace {
         // with specialized emitters above, enum constructors use
         // the AST-level abstraction pattern, to ensure that function
         // types in payloads are re-abstracted correctly.
+#if !defined(_MSC_VER) || defined(__clang__)
         assert(!AssumedPlusZeroSelf);
```

Comment:
For now, I'm going to revert this change. It's possible this is related to my changes elsewhere in my working branch. However, in previous PRs I have seen assertions happening as they assume a define left to right ordering of method results passed to a method, so could be something more